zerver/webhooks/front/tests.py

```diff
@@ -0,0 +1,282 @@
+from typing import Text
+import ujson
+
+from zerver.lib.test_classes import WebhookTestCase
+
+class FrontHookTests(WebhookTestCase):
+    STREAM_NAME = 'front'
+    URL_TEMPLATE = "/api/v1/external/front?&api_key={api_key}"
+    FIXTURE_DIR_NAME = 'front'
+
+    def _test_no_message_data(self, fixture_name: Text) -> None:
+        payload = self.get_body(fixture_name)
+        payload_json = ujson.loads(payload)
+        del payload_json['conversation']['subject']
+        result = self.client_post(self.url, ujson.dumps(payload_json),
+                                  content_type="application/x-www-form-urlencoded")
+
+        self.assert_json_error(result, "Missing required data")
+
+    def _test_no_source_name(self, fixture_name: Text) -> None:
+        payload = self.get_body(fixture_name)
+        payload_json = ujson.loads(payload)
+        del payload_json['source']['data']['first_name']
+        result = self.client_post(self.url, ujson.dumps(payload_json),
+                                  content_type="application/x-www-form-urlencoded")
+
+        self.assert_json_error(result, "Missing required data")
+
+    def _test_no_target_name(self, fixture_name: Text) -> None:
+        payload = self.get_body(fixture_name)
+        payload_json = ujson.loads(payload)
+        del payload_json['target']['data']['first_name']
+        result = self.client_post(self.url, ujson.dumps(payload_json),
+                                  content_type="application/x-www-form-urlencoded")
+
+        self.assert_json_error(result, "Missing required data")
+
+    def _test_no_comment(self, fixture_name: Text) -> None:
+        payload = self.get_body(fixture_name)
+        payload_json = ujson.loads(payload)
+        del payload_json['target']['data']['body']
+        result = self.client_post(self.url, ujson.dumps(payload_json),
+                                  content_type="application/x-www-form-urlencoded")
+
+        self.assert_json_error(result, "Missing required data")
+
+    def _test_no_tag(self, fixture_name: Text) -> None:
+        payload = self.get_body(fixture_name)
+        payload_json = ujson.loads(payload)
+        del payload_json['target']['data']['name']
+        result = self.client_post(self.url, ujson.dumps(payload_json),
+                                  content_type="application/x-www-form-urlencoded")
+
+        self.assert_json_error(result, "Missing required data")
+
+    def test_no_event_type(self) -> None:
+        payload = self.get_body('1_conversation_assigned_outbound')
+        payload_json = ujson.loads(payload)
```

Comment:
Also, I think you should rename all of the fixtures to not start with a number. So `conversation_assigned_outbound` instead of `1_conversation_assigned_outbound`. We don't use any number prefixes anywhere in the fixture file names for our other integrations, so let's stick to that convention. :)